support for Membrane users

skurfer opened the pull request:
These are some small fixes we made (and one not so small) to get things working with Membrane users.

Commit:
compare DateTimes instead of strings

the string value might include time and timezone

Commit:
enumerate users more reliably

Commit:
monkey patch MembraneUserManager

frisi commented on collective/pwexpiry/patches.py:
```diff
@@ -8,6 +8,7 @@
 from plone.app.users.browser.register import BaseRegistrationForm
 from Products.CMFCore.utils import getToolByName
 from Products.CMFPlone.RegistrationTool import RegistrationTool
+from Products.membrane.plugins.usermanager import MembraneUserManager
```
this will break instances w/o Products.membrane installed

skurfer replied:
Good point. We’ll look at a better way.

frisi commented on collective/pwexpiry/patches.py:
```diff
@@ -150,3 +151,45 @@ def authenticateCredentials(self, credentials):
             "ZODBUserManager.authenticateCredentials")
+
+
+MembraneUserManager.original_authenticateCredentials = MembraneUserManager.authenticateCredentials
```
please use collective.monkeypatcher for adding new patches

frisi commented on collective/pwexpiry/pwexpiry_plugin.py:
```diff
@@ -76,7 +76,7 @@ def authenticateCredentials(self, credentials):
 
         password_date = user.getProperty('password_date', '2000/01/01')
-        if str(password_date) != '2000/01/01':
+        if password_date != DateTime('2000/01/01'):
```
as we are storing passwort_date with type date in `memberdata_properties.xml`

``` xml
<property name="password_date" type="date">2000/01/01</property>
```

this looks more sane to me

unfortunately we don't have any tests yet

frisi commented on collective/pwexpiry/scripts/notify_and_expire.py:
```diff
@@ -55,7 +55,7 @@ def notify_and_expire():
     current_time = portal.ZopeTime()
     local_tz = current_time.timezone()
-    for user_id in portal.acl_users.source_users.getUserIds():
+    for user_id in portal.acl_users.getUserIds():
```
don't know enough about acl_users.
can you explain why this change is necessary and if it will work on installations w/o membrane too.

Commit:
only monkey patch MembraneUserManager if available

And only patch one user manager. The goal is to add notifications, but we don't want to fire them twice.
